### PR TITLE
sound making

### PR DESCRIPTION
**Pull request by tejchh**
description:

the beggining of sound.

### CODE REVIEW

File: src/Sounder/Sounder.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        setNote(availableNotes[random])
+    }
+
+    return (
+        <div className='flex-column align-center'>
+            <div className='flex align-center '>
+                <header className='text-purple-400 px-3 flex-auto'>long notes -{'>'}</header>
+                <div id='length-switch flex-auto' className='p-0'>
+                    <Switch
+                        checked={long}
```
Comment (devinjameson) [inline src/Sounder/Sounder.tsx:47]:
```typescript
const isChecked = length === "Long"

checked={isChecked}
```

File: src/Sounder/Sounder.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                        onChange={setLength}
```
Comment (devinjameson) [inline src/Sounder/Sounder.tsx:48]:
```typescript
const handleOnChangeLength = () => {
  setLength(prev => !prev)
}

onChange={handleOnChangeLength}
```

File: src/Sounder/Sounder.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                        className={`${long ? 'bg-purple-900' : 'bg-purple-700'} relative inline-flex h-[38px] w-[74px] shrink-0 cursor-pointer rounded-full border-2 border-gray-400 transition-colors duration-200 ease-in-out focus:outline-none focus-visible:ring-2  focus-visible:ring-white focus-visible:ring-opacity-0 flex-auto`}
```
Comment (devinjameson) [inline src/Sounder/Sounder.tsx:49]: `className={switchClassName}`

File: src/Sounder/Sounder.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    const handleLength = (longness: boolean): string => {
+        if (!longness) {
+            return "50n"
+        }
+        else {
+            return "8n"
+        }
+    }
```
[the remark this answers is not in the document]
Comment (devinjameson) [reply src/Sounder/Sounder.tsx:34]:
```typescript
type NoteLength = "Short" | "Medium" | "Long" | "SuperDuperLong"

const all: NoteLength[] = ["Short", "Medium", "Long", "SuperDuperLong"]

const show = (noteLength: NoteLength): string => {}

const icon = (noteLength: NoteLength): string => {}

const [currentNoteLength, setCurrentNoteLength] = useState<NoteLength.NoteLength>("Short")

<div>
  {NoteLength.all(noteLength => {
    const isActive = noteLength === currentNoteLength
    return <NoteLengthButton noteLength={noteLength} isActive={isActive}/>
  })}
</div>
```

File: src/Sounder/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export { default as Sounder } from "./Sounder"
```
Comment (devinjameson) [inline src/Sounder/index.ts:1]: `export { default } from "./Sounder"`

**Review by devinjameson — APPROVED** [review]
Awesome work! Few thoughts.

File: src/App.css [hunk 6]
```diff
@@ -9,7 +9,7 @@
 
 @media (prefers-reduced-motion: no-preference) {
   .App-logo {
-    animation: App-logo-spin infinite 5s ease-in-out;
+    /*animation: App-logo-spin infinite 5s ease-in-out;*/
```
Comment (johnschoeman) [inline src/App.css:12]: Should this comment be removed?

File: src/App.css [hunk 7]
```diff
@@ -21,18 +21,18 @@
...
-  color: white;
+  color: rgb(108, 34, 34);
 }
 
 .App-link {
   color: #61dafb;
 }
 
-@keyframes App-logo-spin {
+/* @keyframes App-logo-spin {
```
Comment (johnschoeman) [inline src/App.css:31]: Same here?

File: src/Sounder/Sounder.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    const availableNotes = ["C4", "D4", "E4", "F4", "G4", "A4", "B4"]
+    const [long, setLength] = useState(false)
+    const [currentNote, setNote]=useState("C4")
+
+    const handleSound = (): void => {
+        handleNote()
+        synth.triggerAttackRelease(currentNote, handleLength(long))
+    }
+
+    const handleLength = (longness: boolean): string => {
```
Comment (johnschoeman) [inline src/Sounder/Sounder.tsx:27]: It's convention to use the 'is' prefix for Boolean variables. Eg 'isLong'

**Review by johnschoeman — APPROVED** [review]
Nice work! I agree with all of Devin's points. Might be nice to chat about the difference between the view layer and the data layer for an application at some point.

File: .prettierrc.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (johnschoeman) [inline .prettierrc.json:1]: 🥇

File: src/Sounder/Sounder.tsx [hunk 10]
```diff
@@ -1,68 +1,72 @@
...
+    synth.triggerAttackRelease(Note.getNote(randomNote), handleLength(length))
+    startTransition(() => {
+      setCurrentNote(randomNote)
+    })
+  }
 
-    const handleNote =() : void =>{
-        const random = Math.floor(Math.random() * availableNotes.length);
-        setNote(availableNotes[random])
+  const handleLength = (longness: boolean): string => {
```
Comment (johnschoeman) [inline src/Sounder/Sounder.tsx:29]: A few things not following conventions here:

1. Boolean variables are prefixed with `is`, so, while not as cool sounding, this should be be `isLong`

2. `handle` is usually reserved for callbacks for events or side effects, `handleOnClick` `handleReceivedResponse`, etc. as where this more of a business logic conditional based on some state. so I might use the name `determineLoudnessDuration` or something for readability.

In fact you might want to introduce a type for this concept as 'Loudness' seems to be a domain concept for you here. (the lack of which maybe why you have these code smells in the first place) e.g.

```typescript
type Loudness = "Loud" | "Quite"

const toDurationString = (loudness: Loudness): string => {
  switch (loudness) {
    case "Loud":
      return "50n"
    case "Quite"
      return "8n"
  }
}
```

Which may seem like a bit of boiler plate, but by naming the concept accurately in your type system and avoiding primitives, like `boolean`, you leave yourself open for extension and closed to conflation.

e.g. consider how you might introduce a new loudness value, "Medium", in the typed case, vs. the case where you are modeling the concept with `true` and `false` only.

File: src/Sounder/Sounder.tsx [hunk 11]
```diff
@@ -1,68 +1,72 @@
...
+        <header className="text-purple-400 px-3 flex-auto">
+          long notes -{">"}
+        </header>
+        <div id="length-switch flex-auto" className="p-0">
+          <Switch
+            checked={length}
+            onChange={setLength}
+            className={`${
+                length ? "bg-purple-900" : "bg-purple-700"
+            } relative inline-flex h-[38px] w-[74px] shrink-0 cursor-pointer rounded-full border-2 border-gray-400 transition-colors duration-200 ease-in-out focus:outline-none focus-visible:ring-2  focus-visible:ring-white focus-visible:ring-opacity-0 flex-auto`}
```
Comment (johnschoeman) [inline src/Sounder/Sounder.tsx:49]: this is getting a little hard to read. you might consider pulling out some constants:

```typescript
const baseStyle = "relative inline-flex ..."
const switchStyle = length ? `bg-purple-900 ${baseStyle}` : `bg-purple-700 ${baseStyle}`

...

<Switch className={switchStyle} ... />
```

File: src/model/note.ts [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+  noteAccidental: "b" | "" | "#"
+  noteOctave: "1" | "2" | "3" | "4" | "5"
+  noteLength: "Whole" | "Half" | "Quarter" | "Eighth"
+}
+
+export const getNote = (thisNote: Note): string => {
+  return thisNote.noteLetter + thisNote.noteAccidental + thisNote.noteOctave
+}
+
+export const initNote = (noteArr: Note[]): Note => {
```
Comment (johnschoeman) [inline src/model/note.ts:12]: conventionally this function would be called `first` or `head`

File: src/model/note.ts [hunk 13]
```diff
@@ -0,0 +1,14 @@
+export type Note = {
+  noteLetter: "A" | "B" | "C" | "D" | "E" | "F" | "G"
+  noteAccidental: "b" | "" | "#"
+  noteOctave: "1" | "2" | "3" | "4" | "5"
+  noteLength: "Whole" | "Half" | "Quarter" | "Eighth"
+}
+
+export const getNote = (thisNote: Note): string => {
```
Comment (johnschoeman) [inline src/model/note.ts:8]: conventionally this function would be called `show`

File: src/model/scales.ts [hunk 14]
```diff
@@ -0,0 +1,77 @@
+import * as Note from "./note"
+
+//this file is incomplete, temporarily letting this go so i can do real work
```
Comment (johnschoeman) [inline src/model/scales.ts:3]: ❤️ 

Your breakable toy _is_ real work though!

File: src/model/scales.ts [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+  noteOctave: "4",
+  noteLength: "Whole",
+})
+chromaticScale.push({
+  noteLetter: "B",
+  noteAccidental: "",
+  noteOctave: "4",
+  noteLength: "Whole",
+})
+
```
Comment (johnschoeman) [inline src/model/scales.ts:71]: you may want something like

```typescript
type Scale = "Amajor" | "Aminor" | "Bmajor" | etc
```

here.

**Review by johnschoeman — APPROVED** [review]
Looking solid, a few thoughts.

**Pull request merged**